src/Microsoft.AspNet.Mvc.ModelBinding/Validation/SimpleTypesExcludeFilter.cs

```diff
@@ -19,22 +19,15 @@ public bool IsTypeExcluded(Type type)
         {
             Type[] actualTypes;
 
-            var enumerable = type.ExtractGenericInterface(typeof(IEnumerable<>));
-            if (enumerable == null)
+            // The following special case is for <KeyValuePair<K,V>
```

Comment:
The comment should not say what is being done but why it is being done. Did you check what  happens for a dictionary?